File: arma-demo.web/ClientApp/app/models/note.ts
import { User } from './user';
import { Category } from './category';
import { IFilter } from '../interfaces/ifilter';

export class Note implements IFilter {
    id: number;
    title: string;
    value: string;
    isDeleted: boolean;
    user: User;
    category: Category;

    get filter(): string { return this.title }

    constructor() {
        this.user = new User();
        this.category = new Category();
    }
}

File: arma-demo.data/Note.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace arma_demo.data
{
    public class Note
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string Title { get; set; }
        public string Value { get; set; }
        public bool IsDeleted { get; set; }

        public User User { get; set; }
    }
}
